SDL/Constants.cuh

```diff
@@ -1,78 +1,122 @@
 #ifndef Constants_cuh
 #define Constants_cuh
 
+#include <alpaka/alpaka.hpp>
+#include "../code/alpaka_interface/CachedBufAlloc.h"
+
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
 #include <cuda_fp16.h>
+#endif
 
-#ifdef FP16_Base //This changes pT5 and pT3  and T3 completely. T5 for non regression parameters
-#define __F2H __float2half  
-#define __H2F __half2float  
-typedef __half FPX;
+// Half precision wrapper functions, turned off.
+#if defined(FP16_Base)
+#define __F2H //__float2half
+#define __H2F //__half2float
+typedef /*__half*/ float FPX;
 #else
 #define __F2H
 #define __H2F
-typedef float FPX; 
-#endif
-#ifdef FP16_T5 // changes T5 regression values
-#define __F2H_T5 __float2half  
-#define __H2F_T5 __half2float  
-typedef __half FPX_T5;
-#else
-#define __F2H_T5
-#define __H2F_T5
-typedef float FPX_T5; 
+typedef float FPX;
 #endif
-#ifdef FP16_dPhi // changes segment dPhi values
-#define __F2H_dPhi __float2half  
-#define __H2F_dPhi __half2float  
-typedef __half FPX_dPhi;
-#else
-#define __F2H_dPhi
-#define __H2F_dPhi
-typedef float FPX_dPhi; 
-#endif
-#ifdef FP16_circle // changes segment circle values
-#define __F2H_circle __float2half  
-#define __H2F_circle __half2float  
-typedef __half FPX_circle;
-#else
-#define __F2H_circle
-#define __H2F_circle
-typedef float FPX_circle; 
+
+using Idx = std::size_t;
+using Dim = alpaka::DimInt<3u>;
+using Dim1d = alpaka::DimInt<1u>;
+using Vec = alpaka::Vec<Dim,Idx>;
+using Vec1d = alpaka::Vec<Dim1d,Idx>;
+using QueueProperty = alpaka::NonBlocking;
+using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+
+Vec const elementsPerThread(Vec::all(static_cast<Idx>(1)));
+
+// - AccGpuCudaRt
+// - AccCpuThreads
+// - AccCpuFibers
+// - AccCpuSerial
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    using Acc = alpaka::AccGpuCudaRt<Dim, Idx>;
+#elif ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLED
+    using Acc = alpaka::AccCpuThreads<Dim, Idx>;
+#elif ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLED
+    using Acc = alpaka::AccCpuFibers<Dim, Idx>;
+#elif ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+    using Acc = alpaka::AccCpuSerial<Dim, Idx>;
 #endif
-#ifdef FP16_seg // changes segment values
-#define __F2H_seg __float2half  
-#define __H2F_seg __half2float  
-typedef __half FPX_seg;
-#else
-#define __F2H_seg
-#define __H2F_seg
-typedef float FPX_seg; 
+
+#ifndef ALPAKA_ACC_GPU_CUDA_ENABLED
+struct uint4
```

Comment:
Where is this used? Maybe some comment would help, because it seems very, very generic.

Reply:
This is only needed because we are compiling LST.cc and sdl.cc with g++, and it includes some of the .cu files and .cuh files. uint4 is defined by CUDA, so I needed to have a generic implementation to not through compilation errors from the g++ stuff. This will become more important when we try to run the code on the CPU, because we need a uint4 implementation that mimics what our code does now on the GPU or we need to make changes and remove uint4 from the algo.

Reply:
So, this is some kind of placeholder for the CPU implementation then? I am fine with it staying in with a comment added, thanks!

Reply:
Yup, pretty much.